.github/actions/install-llvm/action.yml

```diff
@@ -0,0 +1,66 @@
+name: install-llvm
+description: "Builds and installs LLVM from source using install-llvm-from-source tasks"
+
+inputs:
+  version:
+    description: "The version of LLVM to install."
+    required: true
+  os:
+    description: "The OS being built upon."
+    required: true
+  directory:
+    description: "The directory to install LLVM binaries to."
+    required: true
+  target:
+    description: "The build script target."
+    required: true
+  arch:
+    description: "TODO: The OS arch being built for."
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: LLVM install cache
+      id: cache-llvm
+      uses: actions/cache@v3
+      with:
+        path: ${{ inputs.directory }}
+        key: llvm-${{ inputs.version }}-${{ inputs.arch }}-${{ inputs.os }}-${{ inputs.target }}-${{ hashFiles('Cargo.lock') }}
+        restore-keys: llvm-${{ inputs.version }}-${{ inputs.arch }}-${{ inputs.os }}-${{ inputs.target }}
+
+    - name: Linux - Install build dependencies, ninja
+      run: sudo apt-get install -y ninja-build
+      shell: pwsh
+      if: ${{ (inputs.os == 'ubuntu-20.04') && (steps.cache-llvm.outputs.cache-hit != 'true') }}
+    - name: Windows - Install build dependencies, ninja
+      run: choco install --accept-license -y ninja
+      shell: pwsh
+      if: ${{ (inputs.os == 'windows-2019') && (steps.cache-llvm.outputs.cache-hit != 'true') }}
+    - name: MacOS - Install build dependencies, ninja
+      run: brew install ninja
+      shell: pwsh
+      if: ${{ (inputs.os == 'macos-11') && (steps.cache-llvm.outputs.cache-hit != 'true') }}
+    - name: Windows - Install LLVM 14.0.6
+      run: choco install llvm --version=14.0.6 --allow-downgrade
```

Comment:
Why does this ignore inputs.version?

Reply:
That is the clang tooling used to build LLVM, not the LLVM version to build

Reply:
I think it'd be less confusing if this was in the earlier "Windows - Install build dependencies" step.

Reply:
Did some retesting and llvm on windows is no longer needed in the builds. I now uninstall it from the images.